frontend/main-app/src/app/App.test.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import {render, screen} from '@testing-library/react';
+import {App} from './App';
+import {MemoryRouter} from "react-router";
+
+
+describe('App component', () => {
+    it("renders Login when accessing '/'", () => {
+        render(
+            <MemoryRouter>
+                <App/>
+            </MemoryRouter>
+        )
+
+        const title = screen.getByRole("heading", {name: "Log in"});
+
+        expect(title).toBeInTheDocument();
+    })
+
+    it("renders Login when accessing '/login'", () => {
+        render(
+            <MemoryRouter initialEntries={['/login']}>
+                <App/>
+            </MemoryRouter>
+        )
+
+        const title = screen.getByRole("heading", {name: "Log in"});
+
+        expect(title).toBeInTheDocument();
+    })
+
+    it("renders Signup when accessing '/signup'", () => {
+        render(
+            <MemoryRouter initialEntries={['/signup']}>
+                <App/>
+            </MemoryRouter>
+        )
+
+        const title = screen.getByRole("heading", {name: "Sign up"});
+
+        expect(title).toBeInTheDocument();
+    })
+
+    it("renders Form Builder when accessing '/form-builder'", () => {
+        render(
+            <MemoryRouter initialEntries={['/form-builder']}>
+                <App/>
+            </MemoryRouter>
+        )
+
+        const title = screen.getByRole("heading", {name: "Form Builder"});
+
+        expect(title).toBeInTheDocument();
+    })
+
+    it("renders footer and header when accessing '/'", () => {
+        render(
+            <MemoryRouter>
+                <App/>
+            </MemoryRouter>
+        );
+
+        const header = screen.getByRole("banner");
+        const footer = screen.getByRole("contentinfo");
+
+        expect(header).toHaveTextContent("Form Factory");
+        expect(footer).toHaveTextContent("Designsystemet");
+    })
+
+    it("renders footer and header when not accessing '/'", () => {
+        render(
+            <MemoryRouter initialEntries={['/form-builder']}>
+                <App/>
+            </MemoryRouter>
+        );
+
+        const header = screen.getByRole("banner");
+        const footer = screen.getByRole("contentinfo");
+
+        expect(header).toHaveTextContent("Form Factory");
+        expect(footer).toHaveTextContent("Designsystemet");
+    })
+});
```

Comment:
As you are using the ```render``` with the ```MemoryRouter``` and ```App``` so many times, I would suggest to refactor the function into its own separate function, something like this: 

```tsx
const renderApp = (initialEntries?: string[]) => {
        render(
            <MemoryRouter initialEntries={initialEntires && initialEntries}>
                <App/>
            </MemoryRouter>
        );
}
```

Then you can call it like this in the tests: 
```tsx
renderApp(); 
```
or 
```tsx
renderApp(['/form-builder']);
```

Reply:
Good point!
Function added. I tried using just `<MemoryRouter initialEntries={initialEntires}>` instead of `<MemoryRouter initialEntries={initialEntires && initialEntries}>`, and this also works. I guess the difference is if the initialEntries prop recieves `undefined` or `false`. Is there a best practice for this?

Reply:
In this case I do not think it makes any difference to be honest 😄

Reply:
In this case I do not think it makes any difference 😄